src/OpenCensus.Exporter.Stackdriver/Implementation/StackdriverMetricsExporterWorker.cs

```diff
@@ -0,0 +1,261 @@
+
+        private object locker = new object();
+
+        public StackdriverMetricsExporterWorker(
+           IViewManager viewManager,
+           StackdriverStatsConfiguration configuration)
+        {
+            GaxPreconditions.CheckNotNull(configuration, "configuration");
+            GaxPreconditions.CheckNotNullOrEmpty(configuration.ProjectId, "configuration.ProjectId");
+
+            this.viewManager = viewManager;
+            monitoredResource = configuration.MonitoredResource;
+            project = new ProjectName(configuration.ProjectId);
+            metricServiceClient = MetricServiceClient.Create();
+            domain = GetDomain(configuration.MetricNamePrefix);
+            displayNamePrefix = GetDisplayNamePrefix(configuration.MetricNamePrefix);
+        }
+
+        public void Start()
+        {
+            lock (locker)
+            {
+                if (!isStarted)
+                {
+                    tokenSource = new CancellationTokenSource();
+
+                    Task.Factory.StartNew(DoWork, tokenSource.Token);
+
+                    isStarted = true;
+                }
+            }
+        }
+
+        public void Stop()
+        {
+            lock (locker)
+            {
+                if (!isStarted)
+                {
+                    return;
+                }
+
+                tokenSource.Cancel();
+            }
+        }
+
+        /// <summary>
+        /// Periodic operation happening on a dedicated thread that is 
+        /// capturing the metrics collected within a collection interval
+        /// </summary>
+        private void DoWork()
+        {
+            try
+            {
+                TimeSpan sleepTime = collectionInterval;
+                var stopWatch = new Stopwatch();
+
+                while (!tokenSource.IsCancellationRequested)
+                {
+                    // Calculate the duration of collection iteration
+                    stopWatch.Start();
+
+                    // Collect metrics
+                    Export();
+
+                    stopWatch.Stop();
+
+                    // Adjust the wait time - reduce export operation duration
+                    sleepTime = collectionInterval.Subtract(stopWatch.Elapsed);
+                    sleepTime = sleepTime.Duration();
+
+                    // If the cancellation was requested, we should honor
+                    // that within the cancellation interval, so we wait in 
+                    // intervals of <cancellationInterval>
+                    while (sleepTime > cancellationInterval && !tokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(cancellationInterval);
+                        sleepTime = sleepTime.Subtract(cancellationInterval);
+                    }
+
+                    Thread.Sleep(sleepTime);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
```

Comment:
yeah... we will need to clean up error reporting

Reply:
Yeah, we should decide on how to log in OpenCensus library and apply it to exporters too.